src/wallet/gtest/test_wallet.cpp

```diff
@@ -2149,3 +2171,47 @@ TEST(WalletTests, SaplingNoteLocking) {
     EXPECT_FALSE(wallet.IsLockedNote(sop1));
     EXPECT_FALSE(wallet.IsLockedNote(sop2));
 }
+
+TEST(WalletTests, GenerateUnifiedAddress) {
+    SelectParams(CBaseChainParams::TESTNET);
+    TestWallet wallet(Params());
+
+    UAGenerationResult uaResult = wallet.GenerateUnifiedAddress(0, diversifier_index_t(0), {ReceiverType::P2PKH, ReceiverType::Sapling});
+
+    // If the wallet does not have a mnemonic seed available, it is
+    // treated as if the wallet is encrypted.
```

Comment:
Consider adding an assertion (I'm not sure where) that we can't get into that state for an initialized mainnet or testnet wallet.